File: GuiaApe06/Ejercicio2.cpp
#include <iostream>   // Librería para entrada y salida de datos
#include <vector>     // Librería para usar vectores dinámicos

using namespace std;

int main() {

    // Crear un vector para almacenar notas decimales
    vector<double> notas;

    // Variable auxiliar para guardar cada nota ingresada
    double nota;

    // Ciclo para ingresar 6 notas
    for(int i = 0; i < 6; i++) {

        // Pedir una nota al usuario
        cout << "Ingrese nota: ";

        // Leer la nota
        cin >> nota;

        // Agregar la nota al vector
        notas.push_back(nota);
    }

    // Mostrar título
    cout << "\nNotas registradas:\n";

    // Ciclo para mostrar las notas guardadas
    for(int i = 0; i < notas.size(); i++) {

        // Mostrar cada nota
        cout << notas[i] << endl;
    }

    return 0;
}

File: GuiaApe06/Ejercicio2.java
import java.util.ArrayList; // Librería para usar ArrayList
import java.util.Scanner;   // Librería para entrada de datos

public class Ejercicio2 {

    public static void main(String[] args) {

        // Crear objeto Scanner para leer datos
        Scanner sc = new Scanner(System.in);

        // Crear un ArrayList para guardar notas decimales
        ArrayList<Double> notas = new ArrayList<>();

        // Variable auxiliar para almacenar cada nota
        double nota;

        // Ciclo para ingresar 6 notas
        for(int i = 0; i < 6; i++) {

            // Pedir una nota al usuario
            System.out.print("Ingrese nota: ");

            // Leer la nota ingresada
            nota = sc.nextDouble();

            // Agregar la nota al ArrayList
            notas.add(nota);
        }

        // Mostrar título
        System.out.println("\nNotas registradas:");

        // Recorrer el ArrayList y mostrar las notas
        for(Double n : notas) {

            // Mostrar cada nota
            System.out.println(n);
        }

        // Cerrar Scanner
        sc.close();
    }
}
